Tasks/CopyFilesToMachines/CopyFilesToMachines.ps1

```diff
@@ -160,7 +109,7 @@ else
 	
         Write-Output "Copy will be Started for - $machine"
 
-        $job = Start-Job -ScriptBlock $CopyJob -ArgumentList $environmentName, $envOperationId, $machine, $sourcePath, $targetPath, $machineUserName, $machinePassword, $cleanTargetBeforeCopy
+        $job = Start-Job -ScriptBlock $CopyJob -ArgumentList $environmentName, $envOperationId, $machine, $sourcePath, $targetPath, $machineUserName, $machinePassword, $cleanTargetBeforeCopy, $connection
 
         $Jobs.Add($job.Id, $machine)
     
```

Comment:
I would recommend to have the Remove-job right after Recieve-Jobm in that way later if any code in between goes awry, we know we are not stuck in an infinite loop. Just some good coding practice.